interfaces/mount/entry.go

```diff
@@ -0,0 +1,105 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package mount
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Entry describes an /etc/fstab-like mount entry.
+//
+// Fields are named after names in struct returned by getmntent(3).
+//
+// struct Mntent {
+//     char *Mnt_fsname;   /* name of mounted filesystem */
+//     char *Mnt_dir;      /* filesystem path prefix */
+//     char *Mnt_type;     /* mount type (see Mntent.h) */
+//     char *Mnt_opts;     /* mount options (see Mntent.h) */
+//     int   Mnt_freq;     /* dump frequency in days */
+//     int   Mnt_passno;   /* pass number on parallel fsck */
+// };
+type Entry struct {
+	FsName  MntFsName
+	Dir     MntDir
+	FsType  MntFsType
+	Options MntOptions
```

Comment:
This feels over-typed.  Can we just have string/[]string on those, and a single `Entry.String` method?

Also, this feels neither aligned with the original names, nor nice to read. Going one way or the other seems fine, but the middle is a bit awkward.

For example, this reads nicely:

```
type Entry struct {
        Name    string
        Dir     string
        Type    string
        Options []string

        DumpFrequency   int
        CheckPassNumber int
}
```

While this reads not as nicely, but is aligned with the original code:
```
type Entry struct {
        Fsname string
        Dir    string
        Type   string
        Opts   []string
        Freq   int
        Passno int
}
```

Either seem fine and better than different and mangled.